opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -580,880 +354,418 @@ class Init : private boost::noncopyable
-                unitName_ = "SM3/DAY";
-                targetRateToSiConversionFactor_ = cubic(meter)/day; // m^3/s -> m^3/day
-            }
-            else
-                OPM_THROW(std::logic_error, "Deck uses unexpected unit system");
-        }
-        else
-            OPM_THROW(std::logic_error,
-                      "Unexpected phase " << phase);
-        return unitName_;
+    if( !compressed_to_cartesian ) {
+        /* without a global to active mapping we assume identity mapping, i.e.
+         * 0 -> 0, 1 -> 1 etc.
+         */
+        fun::iota range( num_cells );
+        std::copy( range.begin(), range.end(), this->global_to_active.begin() );
+        return;
     }
 
-    const char* unitName_;
-    double targetRateToSiConversionFactor_;
-};
-
-/// Monitors the total production in a well.
-class WellTotal : public WellReport
-{
-public:
-    WellTotal(const Summary& summary,
-              Opm::EclipseStateConstPtr eclipseState,
-              Opm::WellConstPtr well,
-              PhaseUsage uses,
-              BlackoilPhases::PhaseIndex phase,
-              WellType type,
-              UnitSystem::UnitType unitType)
-        : WellReport(summary,
-                     eclipseState,
-                     well,
-                     uses,
-                     phase,
-                     type,
-                     'T',
-                     handleUnit_(phase, unitType))
-          // nothing produced when the reporting starts
-        , total_(0.)
-    { }
-
-    virtual double retrieveValue(const int writeStepIdx,
-                                 const SimulatorTimerInterface& timer,
-                                 const WellState& wellState,
-                                 const std::map<std::string, int>& wellNameToIdxMap)
-    {
-        if (writeStepIdx == 0) {
-            // We are at the initial state.
-            // No step has been taken yet.
-            return 0.0;
-        }
-
-        if (well_->getStatus(timer.reportStepNum()) == WellCommon::SHUT) {
-            // well is shut in the current time step
-            return 0.0;
-        }
-
-        // find the index for the quantity in the wellState
-        this->updateTimeStepWellIndex_(wellNameToIdxMap);
-        if (this->flatIdx_ < 0) {
-            // well not active in current time step
-            return 0.0;
-        }
-
-        // due to using an Euler method as time integration scheme, the well rate is the
-        // average for the time step. For more complicated time stepping schemes, the
-        // integral of the rate is not simply multiplying two numbers...
-        const double intg = timer.stepLengthTaken() * rate(wellState);
-
-        // add this timesteps production to the total
-        total_ += intg;
-        // report the new production total
-        return unit::convert::to(total_, targetRateToSiConversionFactor_);
-    }
+    for( size_t active_index = 0; active_index < num_cells; ++active_index )
```

Comment:
We are maintaning varieties of this mapping again and again; for the future we should try to use *only* the grid - all the required mappings are there.